no more impostor comets

belisarius222 opened the pull request:
Only treat an address as a comet if it's bigger than 64 bits.  Otherwise, a comet could impersonate a different kind of ship, which posed security problems.  This PR changes comet generation in Zuse, and comet validation in Ames.  I'm not aware of other places where this validation would be necessary, but it's possible others exist.

ohAitch commented on pkg/arvo/sys/vane/ames.hoon:
```diff
@@ -1056,6 +1056,9 @@
     =+  ;;  =open-packet            (cue signed)
+    ::  assert the comet can't pretend to be a moon or other address
+    ::
+    ?>  (gth 64 (met 0 sndr.packet))
```
Should this also just use `clan:title`? We probably don't want 129-bit "comets" either…

belisarius222 replied:
unnecessary -- `sndr.packet` is read from a 128-bit slot in the packet, not from a jammed noun

ohAitch commented on pkg/arvo/sys/zuse.hoon:
```diff
@@ -2173,6 +2173,74 @@
   --  ::  jael
+::                                                      ::::
+::::                      ++lenz                          ::  (1l) scry network
+  ::                                                    ::::
```
This core at minimum seems unrelated to the commit message

belisarius222 replied:
oops, cross-talk from another line of work -- will remove, apologies

belisarius222 commented:
ok, cleaned up the git snafu

joemfb commented:
FWIW, i'd be really tempted to just write these as `?>  ?=(%pawn (clan:title some-ship))`.

ohAitch commented:
Have the "is a bitwidth check" in the comments and "is a ship class check" in the code instead of the other way around :P

belisarius222 commented:
Ah, yeah I could add this check to `+clan:title` if it isn't implicitly there.  That would be cleaner.

belisarius222 commented:
redid it using `+clan:title`

joemfb commented:
@pilfer-pandex, the same sanity check likely needs to be added to comet "mining" in king-haskell.

ohAitch commented:
I mean, you're not going to mine 64 trailing zero bits on /accident/, this
seems mostly important to validate on the receiving end.

(Really if mining specific addresses were remotely feasible comets wouldn't
make sense as a concept, the worst you could hope for is hitting a random
moon which may or may not be spawned - but with a mature network this could
become, at all a viable grieving route via birthday attacks)

On Tue, Oct 27, 2020 at 19:48, Joe Bryan <notifications@github.com> wrote:

> @pilfer-pandex <https://github.com/pilfer-pandex>, the same sanity check
> likely needs to be added to comet "mining" in king-haskell.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/urbit/urbit/pull/3812#issuecomment-717643267>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAOFPBSMAD2T7D3MEP6KPBLSM5Z7RANCNFSM4TBMQ22A>
> .
>

belisarius222 commented:
@ohAitch Mining a moon-length comet is extremely unlikely, but still invalid, so it's worth checking when trying to create a comet address -- otherwise it's wrong, and this is Kelvin-versioned kernel code that defines the spec.

@philipcmonk I've addressed review comments. PTAL

philipcmonk reviewed (approved):
LGTM
